### ISSUE TITLE
Title: Infinite Fetching Loop and Memory Leak in fetchBooksFromDB Function

### ISSUE THREAD

**Issue by dev-george-nikolaidis** [gbowne1/codebooker#220]
**Describe the bug**
The `fetchBooksFromDB` function triggers an infinite loop of data fetching when used in conjunction with `useEffect`, leading to significant performance degradation and a memory leak.

### Expected Behavior
`fetchBooksFromDB` should fetch the data once and stop, without causing an infinite loop or memory leaks.

### Actual Behavior
The function continuously fetches data without stopping, causing browser performance issues and memory leaks.

### Additional Information
![memoryLeak](https://github.com/gbowne1/codebooker/assets/82717511/26573237-e8e1-4671-a843-28d3b6a1b297)

**Comment by gbowne1** [comment]
This is also due to the fact we were also trying to handle a case for using the data.json file 

After thinking about this from the previous issues we have had with this, I also think we could have:

```js
const fetchBooksFromDB = async () => {
  try {
    setLoading(true);

    // Attempt to fetch data from the database
    const response = await axios.get('http://localhost:3001/api/books/getall');
    const books = response.data;

    // Database success: Use books from the database
    if (books.length > 0) {
      setMyRows([...books]);
      return; // Exit the function if database fetch is successful
    }

    // Handle database failure: Try fetching from data.json
    console.warn('Database fetch failed. Attempting to fetch from data.json');
    const fileData = await fetch('data.json');
    const fileBooks = await fileData.json();

    // Data.json success: Use books from data.json and potentially update the database
    if (fileBooks.length > 0) {
      setMyRows([...fileBooks]);
      // Optionally call a function to add the fetched books to the database (addBooksToDB)
    } else {
      // Both database and data.json failed: Handle fallback scenario (e.g., display error message)
      console.error('Both database and data.json fetch failed.');
    }
  } catch (error) {
    // Handle errors related to both database and data.json fetching
    console.error('Error fetching books:', error);
  } finally {
    setLoading(false);
  }
};
```

**Comment by gbowne1** [comment]
This still has not been solved yet and it really decreses the project speed by a lot by it continually fetching the library.
